zeek

```diff
@@ -18,179 +18,214 @@
+		docker_cmd+=("$IMAGE_NAME")
+
+		echo "Starting the Zeek docker container" >&2
+		"${docker_cmd[@]}"
+
+		# Fix current symlink for the host (sleep to give Zeek time to finish starting)
+		(sleep 30s; docker exec "$CONTAINER_NAME" ln -sfn "$HOST_ZEEK/spool/manager" /usr/local/zeek/logs/current) &
 
-disable)
-	#Command(s) needed to stop the service on future boots
-	echo "Blocking Zeek docker container from starting on future boots" >&2
-	if [ "$CONTAINER_RUNNING" == "false" ]; then
-		echo "Zeek is stopped - please start first to set restart policy." >&2
-		exit 0
-	fi
+		;;
+	stop)
+		#Command(s) needed to stop the service right now
+
+		if [ "$CONTAINER_RUNNING" != "false" ]; then
+			echo "Stopping the Zeek docker container" >&2
+			docker stop -t 90 "$CONTAINER_NAME" >&2
+		else
+			echo "Zeek is already stopped." >&2
+		fi
+		
+		docker rm --force "$CONTAINER_NAME" >/dev/null 2>&1
+		;;
+
+	restart|force-restart)
+		#Command(s) needed to stop and start the service right now
+		#You can test the value of "$action" in case there's a different set of steps needed to "force-restart"
+		echo "Restarting the Zeek docker container" >&2
+		$0 stop
+		$0 start
+		;;
+
+	status)
+		#Command(s) needed to tell the user the state of the service
+		echo "Zeek docker container status" >&2
+		docker ps --filter name=zeek >&2
+
+		echo "Zeek processes status" >&2
+		docker exec "$CONTAINER_NAME" zeekctl status >&2
+		;;
+
+	reload)
+		#Command(s) needed to tell the service to reload any configuration files
+		echo "Reloading Zeek docker container configuration files" >&2
+		#Note; I'm not aware of a way to do a config file reload, so forcing a full restart at the moment.
+		$0 stop
+		$0 start
+		;;
 
-	$SUDO docker update --restart no "$CONTAINER_NAME" >&2
-	;;
+	enable)
+		#Command(s) needed to start the service on future boots
+		echo "Enabling Zeek docker container on future boots" >&2
+		if [ "$CONTAINER_RUNNING" == "false" ]; then
+			echo "Zeek is stopped - please start first to set restart policy." >&2
+			exit 0
+		fi
 
-pull|update)
-	#Command needed to pull down a new version of Zeek if there's a new docker image
-	$SUDO docker pull "$IMAGE_NAME"
+		docker update --restart always "$CONTAINER_NAME" >&2
+		;;
+
+	disable)
+		#Command(s) needed to stop the service on future boots
+		echo "Blocking Zeek docker container from starting on future boots" >&2
+		if [ "$CONTAINER_RUNNING" == "false" ]; then
+			echo "Zeek is stopped - please start first to set restart policy." >&2
+			exit 0
+		fi
+
+		docker update --restart no "$CONTAINER_NAME" >&2
+		;;
+
+	pull|update)
+		#Command needed to pull down a new version of Zeek if there's a new docker image
+		docker pull "$IMAGE_NAME"
 
-	$0 stop
-	$0 start
-	;;
+		$0 stop
+		$0 start
+		;;
+
+	*)
+		echo "Unrecognized action $action , exiting" >&2
+		exit 1
+		;;
+	esac
 
-*)
-	echo "Unrecognized action $action , exiting" >&2
-	exit 1
-	;;
-esac
+	exit 0
+}
 
-exit 0
+if [ "$0" = "$BASH_SOURCE" ]; then
```

Comment:
Only execute the main function if the script was called from the command line and not when sourced as a library.